### PR TITLE
Med 106 implement booking api

### PR DESCRIPTION
**Pull request by SoulOfDarknes**
1. Add create a appointments 
2. Can get appointments from DB
3. Can delete
4. add guards

**PR checklist:**

Common
- [x] types for input and output params
- [x]  no any, should be also added to eslint rules
- [x]  try/catch for any async function
- [x] no magic numbers
- [x]  compare only with constants not with strings
(instead of user === ‘admin’, better user === roles.admin)
- [x]  no ternary operator inside ternary operator (bad example: question ? first:
second ? third: fourth)
- [x]  avoid similar types with duplicating by generics
- [x]  no console.log in pr
- [x]  .env file should be in .gitignore
- [x]  delete commented code if it's not part of documentation
- [x]  no links in the code, env links should be in env file (for example: server url),
constant links (for example default avatar URL) should be in constant file
- [x]  constants and function names should be lowercase.
- [x]  no dates format in the code (like "dd/MM/yyyy”), move it in global constant
variable
- [x]  import rules. imports should come in a specific order: node modules, absolute
path, relative path
- [x]  strings should be in the constant variable. Example: instead of ‘15 3 * * *’,
const EACH_DAY_15h_15min = ‘15 3 * * *’

1.2 Backend
- [x]  swagger for each endpoint
- [x]  less requests to db, all data should be taken with one query
- [x]  public in method use only is use function externally
- [x] use ConfigService instead of process.env

create
image

get all
image

get one by ID
image

### CODE REVIEW

**Review by oleksii-yurchenko — APPROVED** [review]

File: src/modules/appointments/appointments.service.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  async getAllAppointments(): Promise<Appointment[]> {
+    return this.appointmentRepository.find();
+  }
+
+  async getAppointmentById(id: number): Promise<Appointment> {
+    const options: FindOneOptions<Appointment> = { where: { id } };
+    return this.appointmentRepository.findOne(options);
+  }
+
+  async createAppointment(
```
Comment (lyudad) [inline src/modules/appointments/appointments.service.ts:37]: make sure you store data in 0 timezone

Comment (SoulOfDarknes) [reply src/modules/appointments/appointments.service.ts:37]: changed and tested, looks fine now

File: src/modules/appointments/appointments.controller.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  @Get()
+  @ApiOperation({ summary: 'Get all appointments' })
+  @ApiResponse({
+    status: 200,
+    description: 'Returns an array of appointments',
+    type: Appointment,
+    isArray: true,
+  })
+  async getAllAppointments(): Promise<Appointment[]> {
```
Comment (skazzp) [inline src/modules/appointments/appointments.controller.ts]: If this is a route for step1 booking appointment, then it should be restricted to local doctor.
same with create route (check other routes too)

**Review by skazzp — CHANGES_REQUESTED** [review]

File: src/modules/appointments/appointments.controller.ts (hunk 2 above)
Comment (SoulOfDarknes) [reply src/modules/appointments/appointments.controller.ts]: add guards

**Review by skazzp — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**